rest_framework_json_api/views.py

```diff
@@ -62,33 +70,78 @@ class MyViewSet(viewsets.ModelViewSet):
         return qs
 
 
-class AutoPrefetchMixin(object):
+class AutoPreloadMixin(object):
+    """
+    This mixin provides a helper attributes to select or prefetch related models
+    based on the include specified in the URL.
+
+    __all__ can be used to specify a prefetch which should be done regardless of the include
+
+    .. code:: python
+
+    # When MyViewSet is called with ?include=author it will prefetch author and authorbio
+    class MyViewSet(viewsets.ModelViewSet):
+        queryset = Book.objects.all()
+        prefetch_for_includes = {
+            '__all__': [],
+            'category.section': ['category']
+        }
+        select_for_includes = {
+            '__all__': [],
+            'author': ['author', 'author__authorbio'],
+        }
+    """
+
+    def get_select_related(self, include):
+        return getattr(self, 'select_for_includes', {}).get(include, None)
+
+    def get_prefetch_related(self, include):
+        return getattr(self, 'prefetch_for_includes', {}).get(include, None)
+
     def get_queryset(self, *args, **kwargs):
         """ This mixin adds automatic prefetching for OneToOne and ManyToMany fields. """
-        qs = super(AutoPrefetchMixin, self).get_queryset(*args, **kwargs)
+        qs = super(AutoPreloadMixin, self).get_queryset(*args, **kwargs)
         included_resources = get_included_resources(self.request)
 
-        for included in included_resources:
+        for included in included_resources + ['__all__']:
+            # Custom defined "select_related" and "prefetch_related" is a priority
+            select_related = self.get_select_related(included)
+            if select_related is not None:
+                qs = qs.select_related(*select_related)
+                continue
```

Comment:
this might be confusing to user when defining a select and prefetch for some reason that only one of both happens.

also same issue as already [discussed](https://github.com/django-json-api/django-rest-framework-json-api/pull/600#discussion_r268538762)

Best to remove both continue statements. As it is now only one mixin it won't be easy to disable auto prefetching though. After refactoring the method length is quite long so a solution might be to split it up into several method or split it into two mixins again. Maybe there are other solutions but I would want to avoid a flag somehow. Open for suggestions.

Reply:
Spitted into 2 mixins. 
But not actually sure if there are cases when user needs both select_related and prefetch_related and AutoPreload. I guess having AutoPreload by default with ability to override  it (via select or prefetch dict) would work just fine

Reply:
As it is now it is the safest as it behaves as it was before.

Most likely this won't be a problem that auto prefetching and manual preloading is done together as normally manual prefetching is only needed in cases of properties or serializer methods where auto prefetching won't work anyway.